Documentation: automatic tap changer behavior (#1170)
PowerGridModel/power-grid-model: docs/user_manual/calculations.md

| linear search | Start with an initial guess and do a local search with step size 1 for each iteration step.     |
| binary search | Start with a large search region and reduce the search region by half for every iteration step. |

##### Regulatable voltage range outside `u_band`

In the case of an unrealistically set volatage and `u_band`, the viable voltage range regulatable can fall outside
this target band.
This will lead the regulator returning the tap that results in the closest voltage level to the target band.

##### Error type `MaxIterationReached`

> When the regulator can not find a suitable voltage after iterations, error `MaxIterationReached` will be thrown.

Reviewer: Nitpick:
```suggestion
When the regulator cannot find a suitable voltage after all iterations, the MaxIterationReached error is thrown.
```
